src/utils/Store.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { createContext, SetStateAction, useMemo, useState } from 'react';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { ErrorTypes } from '../types/ErrorTypes';
+
+interface Children {
+  children: React.ReactNode;
+}
+
+type ActiveUser = User | null;
+
+interface ActiveUserContextType {
+  activeUser: ActiveUser;
+  setActiveUser: React.Dispatch<SetStateAction<ActiveUser>> | undefined;
+}
+
+export const ActiveUserContext = createContext<ActiveUserContextType>({
+  activeUser: null,
+  setActiveUser: undefined,
+});
+
+export const ActiveUserProvider: React.FC<Children> = ({ children }) => {
+  const [activeUser, setActiveUser] = useState<ActiveUser>(null);
+
+  const value = useMemo(() => ({ activeUser, setActiveUser }), [activeUser]);
+
+  return (
+    <ActiveUserContext.Provider value={value}>
+      {children}
+    </ActiveUserContext.Provider>
+  );
+};
+
+interface Loader {
+  isLoading: boolean;
+  setIsLoading: React.Dispatch<SetStateAction<boolean>> | undefined;
+}
+
+export const LoaderContext = createContext<Loader>({
+  isLoading: false,
+  setIsLoading: undefined,
+});
+
+export const LoaderProvider: React.FC<Children> = ({ children }) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const value = useMemo(
+    () => ({ isLoading, setIsLoading }),
+    [isLoading, setIsLoading],
```

Comment:
You are using 'setIsLoading' in the dependency array of useMemo, which is unnecessary because 'setIsLoading' is a stable function and won't change between renders. Consider removing it from the dependency array.